lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1201,51 +1183,37 @@ void LocalEnforcer::report_subscriber_state_to_pipelined(
 
 void LocalEnforcer::complete_termination(
     SessionMap& session_map, const std::string& imsi,
-    const std::string& session_id, SessionUpdate& session_update) {
+    const std::string& session_id, SessionUpdate& session_updates) {
   // If the session cannot be found in session_map, or a new session has
   // already begun, do nothing.
-  auto it = session_map.find(imsi);
-  if (it == session_map.end()) {
+  SessionSearchCriteria criteria(imsi, IMSI_AND_SESSION_ID, session_id);
+  auto session_it = session_store_.find_session(session_map, criteria);
+  if (!session_it) {
     // Session is already deleted, or new session already began, ignore.
     MLOG(MDEBUG) << "Could not find session for IMSI " << imsi
                  << " and session ID " << session_id
                  << ". Skipping termination.";
-    return;
   }
-  auto& uc = session_update[imsi][session_id];
-  for (auto session_it = it->second.begin(); session_it != it->second.end();
-       ++session_it) {
-    if ((*session_it)->get_session_id() == session_id) {
-      bool terminated = (*session_it)->complete_termination(uc);
-      if (!terminated) {
-        return;  // error is logged in SessionState's complete_termination
-      }
-      auto termination_req = (*session_it)->make_termination_request(uc);
-      auto logging_cb =
-          SessionReporter::get_terminate_logging_cb(termination_req);
-      reporter_->report_terminate_session(termination_req, logging_cb);
-      // Send to eventd
-      if ((*session_it)->is_radius_cwf_session() == false) {
-        events_reporter_->session_terminated(imsi, *session_it);
-      }
-      // We break the loop below, but for extra code safety in case
-      // someone removes the break in the future, adjust the iterator
-      // after erasing the element
-      uc.is_session_ended = true;
-      it->second.erase(session_it--);
```

Comment:
There is not much logic change here, just getting rid of the loop and using the `find_session` function